testplugins/src/main/java/org/spongepowered/test/data/DataFormatTest.java

```diff
@@ -0,0 +1,79 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.test.data;
+
+import com.google.inject.Inject;
+import net.kyori.adventure.text.Component;
+import org.spongepowered.api.command.Command;
+import org.spongepowered.api.command.Command.Parameterized;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.parameter.Parameter;
+import org.spongepowered.api.data.persistence.DataFormats;
+import org.spongepowered.api.data.persistence.StringDataFormat;
+import org.spongepowered.api.entity.living.player.server.ServerPlayer;
+import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.lifecycle.RegisterCommandEvent;
+import org.spongepowered.plugin.PluginContainer;
+import org.spongepowered.plugin.builtin.jvm.Plugin;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+@Plugin("dataformattest")
+public class DataFormatTest {
+
+    private final PluginContainer plugin;
+
+    @Inject
+    public DataFormatTest(final PluginContainer plugin) {
+        this.plugin = plugin;
+    }
+
+    @Listener
+    public void onRegisterCommand(final RegisterCommandEvent<Parameterized> event) {
+        // /dataformattest snbt {id:"stone",Count:3}
+        event.register(this.plugin, Command.builder()
+                        .executionRequirements(cc -> cc.first(ServerPlayer.class).isPresent())
+                        .addParameters(Parameter.choices(StringDataFormat.class, new HashMap<String, StringDataFormat>() {{
+                                    put("hocon", DataFormats.HOCON.get());
+                                    put("json", DataFormats.JSON.get());
+                                    put("snbt", DataFormats.SNBT.get());
+                                }}).key("format").build(),
+                                Parameter.remainingJoinedStrings().key("value").build())
+                        .executor(context -> {
+
+                            StringDataFormat format = context.requireOne(Parameter.key("format", StringDataFormat.class));
+                            String value = context.requireOne(Parameter.key("value", String.class));
+                            try {
+                                context.cause().audience().sendMessage(Component.text(format.write(format.read(value))));
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
```

Comment:
Don't throw `RuntimeExceptions` indiscriminately like this. In the case of commands, yes, it'll be caught, but the result is ugly. Gracefully handle it instead - we have `CommandException` that you should use instead if you must throw an exception.